tools/cli/lib/parselistargs.c

```diff
@@ -107,15 +108,34 @@ TDNFCliParseListArgs(
                   (void**)&pListArgs);
     BAIL_ON_CLI_ERROR(dwError);
 
-    //Should have scope argument (tdnf list <scope> <pkgnamespecs>)
+    pListArgs->nScope = SCOPE_ALL;
+
+    /* scope can be given as an option */
+    for (pSetOpt = pCmdArgs->pSetOpt;
+         pSetOpt;
+         pSetOpt = pSetOpt->pNext)
+    {
+        if(pSetOpt->nType == CMDOPT_KEYVALUE)
+        {
+            dwError = TDNFCliParseScope(
+                          pSetOpt->pszOptName,
+                          &pListArgs->nScope);
+            if(dwError == ERROR_TDNF_CLI_NO_MATCH)
```

Comment:
```
if(dwError != ERROR_TDNF_CLI_NO_MATCH) {
     BAIL_ON_CLI_ERROR(dwError);
}
```
this will save a comparision

Reply:
It would have to be:
```
            if(dwError != ERROR_TDNF_CLI_NO_MATCH)
            {
                BAIL_ON_CLI_ERROR(dwError);
            }
            dwError = 0;
```
I don't see any benefit - same number of comparisons. And we have to reset `dwError` to `0` anyway or the function may return an error.